src/execution_plan/execution_plan_modify.c

```diff
@@ -313,11 +315,41 @@ void ExecutionPlan_BindPlanToOps(ExecutionPlan *plan, OpBase *root) {
 	}
 }
 
-void ExecutionPlan_AppendSubExecutionPlan(ExecutionPlan *master_plan, ExecutionPlan *sub_plan) {
-	if(!master_plan->sub_execution_plans)
-		master_plan->sub_execution_plans = array_new(ExecutionPlan *, 1);
-	if(sub_plan->record_map) raxFree(sub_plan->record_map);
-	sub_plan->record_map = master_plan->record_map;
-	master_plan->sub_execution_plans = array_append(master_plan->sub_execution_plans, sub_plan);
+OpBase *ExecutionPlan_BuildOpsFromPath(ExecutionPlan *plan, const char **vars,
+									   const cypher_astnode_t *path) {
+	// Initialize an ExecutionPlan that shares this plan's Record mapping.
+	ExecutionPlan *rhs_plan = ExecutionPlan_NewEmptyExecutionPlan();
+	rhs_plan->record_map = plan->record_map;
+
+	// If we have bound variables, build an Argument op that represents them.
+	if(vars) rhs_plan->root = NewArgumentOp(plan, vars);
+
+	AST *ast = QueryCtx_GetAST();
+	// Build a temporary AST holding a MATCH clause.
+	AST *rhs_ast = AST_MockMatchPattern(ast, path);
+
+	ExecutionPlan_PopulateExecutionPlan(rhs_plan, NULL);
+
+	AST_MockFree(rhs_ast);
+	QueryCtx_SetAST(ast); // Reset the AST.
+	// Add filter ops to sub-ExecutionPlan.
+	if(rhs_plan->filter_tree) ExecutionPlan_PlaceFilterOps(rhs_plan, NULL);
+
+	// If the temporary execution plan has added new QueryGraph entities,
+	// migrate them to the master plan's QueryGraph.
+	// (This is only necessary for producing EXPLAIN outputs.)
+	QueryGraph_MergeGraphs(plan->query_graph, rhs_plan->query_graph);
```

Comment:
Maybe this call should be part of `ExecutionPlan_BindPlanToOps` ?

Reply:
:+1: Good call!